storage/innobase/include/fil0fil.h

```diff
@@ -288,7 +288,11 @@ struct fil_space_t {
 
   bool is_encrypted;
 
-  bool exclude_from_rotation;
+  // Used by encryption threads to check whether a space was
+  // excluded from encryption/decryption. We use atomic because
+  // this variable can be check by encryption threads without
+  // them acquiring any locks.
+  std::atomic<bool> exclude_from_rotation;
```

Comment:
I feel encryption threads in 8.0 should move to MDLs and get rid of variables/ custom mutexes..

Reply:
That would be 6 months more work ? ...

Reply:
May be :).. what I observed is Two threads are trying to synchronize on tablespace/table operation. Using variables imho is not the proper way.

User threads already synchronize by using MDLs. InnoDB background threads in 8.0 also moved to MDLs. Purge uses MDLs now. Encryption threads should follow the suite.. Not now.. sometime in future..